CIP-0032/README.md

```diff
@@ -115,14 +115,51 @@ However, the other approaches also have some advantages.
 
 Any one of these factors could be important to particular use cases, so it is good to retain the other options.
 
-### Changing the script context
+### The script context
 
-We don't strictly need to change the script context.
-We could use the fall-back approach that we use for the old interface, namely continuing to pretend that datums are always hashes, and providing the mapping from hashes to datums.
-This would be functionally equivalent.
+#### Including information about inline datums
 
-The only advantage of changing it is that it's more honest to the real representation of transactions, and therefore allows scripts to insist on inline datums if they want.
-It's unclear whether this is desirable.
+In principle we do not need to let scripts see whether a datum is inline or not.
+We could pretend that inline datums are non-inline and insert them into the datum witness map.
+
+The underlying question is: do we want scripts to be able to make assertions about whether datums are inline or not?
+There are reasons to want to do this: _not_ using inline datums causes communication issues for users, and so it is quite reasonable that an application developer may want to be able to enforce their use.
+
+Furthermore, as a general principle we try to keep the script context as faithful to real transactions as possible.
+Even if the use for the information is not immediately obvious, we try to err on the side of providing it and letting users decide.
+
+Hence we _do_ include information about inline datums in the script context.
+
+#### Representation and backwards compatibility
+
+There are a couple of options for how to change the representation of the script context to include the new information, and whether to make a backwards compatibility effort for old language versions.
+
+For the new script context:
+
+1. Match the ledger representation as much as possible: change the fields on inputs and outputs to be either a datum hash or a datum.
+2. Try to only have one way to look up datums: put inline datums in the datum witness map and insert their hashes into the corresponding inputs and outputs; optionally add a boolean to inputs and outputs to indicate whether the datum was originally inline.
+
+For backwards compatibility:
+
+1. Don't try to represent inline datums for scripts using old language versions: old scripts simply can't be run in transactions that use inline datums (since we can't represent the information).
```

Comment:
You probably want to make it explicit that the transaction *fails*, not just that they can't be run.

Reply:
This is more explicit earlier, but I'll make it even more explicit.